OpenBoard.pro

```diff
@@ -466,8 +462,8 @@ linux-g++* {
     LIBS += -lX11
 
     greaterThan(QT_MAJOR_VERSION, 5) {
-        LIBS += -lquazip6
-        INCLUDEPATH += "/usr/include/quazip6"
+        LIBS += -lquazip1-qt6
+        INCLUDEPATH += "/usr/include/QuaZip-Qt6-1.4/quazip"
```

Comment:
When I looked again today I found that I was wrong to have said that the Qt5 build was broken by this change, it was already inside a condition block gated on the value of QT_MAJOR_VERSION.

When I tried the cmake build as suggested, I ran it first to a clean checkout of dev without re-applying my changes, and found that for cmake this change is not required - but it is required for a qmake6/make build.

Reply:
> When I looked again today I found that I was wrong to have said that the Qt5 build was broken by this change, it was already inside a condition block gated on the value of QT_MAJOR_VERSION.
> 
> When I tried the cmake build as suggested, I ran it first to a clean checkout of dev without re-applying these changes to OpenBoard.pro, and found that for cmake this change is not required - but both lines need to change for qmake6/make build to work.

Ok, I see. So the question is whether there are Ubuntu versions where quazip6 is installed to the paths used before, and whether your change would break the  build on those versions.

The `cmake` build can cope with that, because it uses the `cmake` or `pkgconfig` mechanisms for finding the correct path, but here in the `qmake` project file the paths are hardcoded.

Reply:
When I was googling around trying to work out why OpenBoard didn't build
against the older quazip package I found a page explaining that 'quazip1'
was intended to be treated as a separate package from the previous 'quazip'
(with 0.X version numbers), installable side by side, so Ubuntu/Debian
should not treat quazip[0] packages as obsolete just because quazip1 was
available.  The follow on from this was that it was OK for quazip1 to have different install
location and library name conventions from quazip[0].

https://github.com/stachenov/quazip/issues/92

On Ubuntu 23.10, the following variants on the quazip package exist:
```
tim@tim-OptiPlex-7450-AIO:~$ apt list libquazip*
Listing... Done
libquazip-doc/mantic,mantic 0.9.1-3 all
libquazip1-qt5-1/mantic 1.4-1 amd64
libquazip1-qt5-dev/mantic 1.4-1 amd64
libquazip1-qt5-doc/mantic,mantic 1.4-1 all
libquazip1-qt6-1/mantic,now 1.4-1 amd64 [installed,automatic]
libquazip1-qt6-dev/mantic,now 1.4-1 amd64 [installed]
libquazip1-qt6-doc/mantic,mantic 1.4-1 all
libquazip5-1/mantic,now 0.9.1-3 amd64 [installed,automatic]
libquazip5-dev/mantic 0.9.1-3 amd64
```

So it looks like libquazip1 has both Qt5 and Qt6 builds, and the libquazip5 is most likely a legacy build for qt5 using the original libquazip[0], but this is just a guess on my part.  It might be that the cleanest way of having a repo which builds against either Qt5 or Qt6 would be to update the Qt5 dependency from libquazip5-dev to libquazip1-qt5-dev, but right now I'd prefer not to bring any Qt5 components onto the machines available to me (at least until the public release of Noble is available).  Of course I don't know how far back libquazip1-qt5-dev is available, so maybe it's cleaner to stay with what already works for older releases.

Reply:
> Of course I don't know how far back libquazip1-qt5-dev is available, so maybe it's cleaner to stay with what already works for older releases.

This can be answered from the package search: https://packages.ubuntu.com/search?keywords=libquazip1

The newer naming conventions are available since Ubuntu 23.04. However, OpenBoard must also be able to be built with 20.04 and 22.04. So we need a solution which coves both without editing the project file. For the include path this is simple: just add both. But for the library name we need to differentiate.

That's the reason why I was using `cmake` and `pkgconfig` mechanisms in the `cmake` builds to discover libraries and include paths. I have also discussed using `pkgconfig` for the `qmake` project file quite some time ago with the project maintainer, but at that time the outcome was to keep it as it is and probably provide instructions on editing the project file for such cases.